plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -75,13 +76,15 @@ export class ElasticSearchSearchEngine implements SearchEngine {
   static async fromConfig({
     logger,
     config,
+    clientOptions = {},
```

Comment:
I think we may want to decouple ourselves a little bit more explicitly from the underlying client/implementation; having the optionality to dynamically choose a client implementation based on configuration (e.g. to be able to support AWS Elasticsearch Service somewhat transparently)...  Or otherwise be able to transparently replace the implementation in the future would be good.

Instead of passing through `clientOptions` directly through from caller to client, what do you think of pulling out some key/common options as config, reading those in, constructing the options, and then passing them in?

Putting it in config also means there's a reliable mechanism for varying the options by environment, which I imagine many of these options may benefit from.

The schema could be defined here: https://github.com/backstage/backstage/blob/master/plugins/search-backend-module-elasticsearch/config.d.ts#L24

Reply:
Sounds good. Let me know if the latest commit is aligned with what you envisioned. I can tidy it up if you have any ideas, but first I wanted to make sure I understand you correctly.